src/app/api/routes/media.py

```diff
@@ -94,7 +94,11 @@ async def upload_media(media_id: int = Path(..., gt=0),
     entry = await check_for_media_existence(media_id, current_device.id)
 
     # Concatenate the first 32 chars (to avoid system interactions issues) of SHA256 hash with file extension
-    bucket_key = f"{hash_content_file(file.file.read())[:32]}.{file.filename.rpartition('.')[-1]}"
+    file_name = f"{hash_content_file(file.file.read())[:32]}.{file.filename.rpartition('.')[-1]}"
+    # Reset byte position of the file (cf. https://fastapi.tiangolo.com/tutorial/request-files/#uploadfile)
+    await file.seek(0)
+    # If files are in a subfolder of the bucket, prepend the folder path
+    bucket_key = prepend_bucket_folder(file_name)
 
     upload_success = await bucket_service.upload_file(bucket_key=bucket_key,
                                                       file_binary=file.file)
```

Comment:
Au passage, can you also fix the return message below?
            detail="The upload did not succeed"
==>     detail="The download did not succeed"

Reply:
Not sure I follow: here we're actually trying to upload. Did you mean in `get_media_image`?

Reply:
Yes,(for some reason I can't comment directly to point this line in the Github interface)

Reply:
Done!